app/model.js

```diff
@@ -119,6 +119,50 @@ const fetchArticleById = (params) =>
     }
 }
 
+const addUserComment = (params, body) =>
+{
+    if(!/[0-9]+/g.test(params))
+    {
+        return Promise.reject({status: 400, msg: 'invalid article ID: not a number'})
+    }
+    return db.query
+    (
+        `
+        SELECT *
+        FROM articles
+        WHERE article_id = ${params}
+        `
+    )
+    .then((response) =>
+    {
+        if(response.rowCount === 0)
+        {
+            return Promise.reject({status: 404, msg: 'invalid article ID: article not found'})
+        }
+        else
+        {
+            return db.query
+            (
+                `
+                INSERT INTO comments (body, author, article_id)
+                VALUES (\'${body.body}\', \'${body.username}\', ${params})
+                `
+            )
+            .then(() =>
+            {
+                return db.query
+                (
+                    `
+                    SELECT *
+                    FROM comments
+                    WHERE body = \'${body.body}\'
+                    AND author = \'${body.username}\'
+                    AND article_id = ${params}
```

Comment:
You are directly inserting values into your SQL query. This leads us open to the risk of SQL injection. We need to be using parametrized queries here and let postgres's server handle the substitution of these values. We have covered this in lectures over the last two weeks and it is very important that you change this.

Reply:
am i not protected from SQL attack on the code on line 124?